From b0ae1ee2392d0612cce7d61842b78640a04b26f0 Mon Sep 17 00:00:00 2001
From: Li Wang <liwang@redhat.com>
Date: Mon, 17 Jun 2024 11:24:00 +0800
Subject: [PATCH 3/3] rpc_svc_1: Fix incompatible pointer type error
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

To get rid of the compile error with GCC-14:

rpc_svc_1.c:109:12: error: assignment to ‘struct sockaddr_in *’ from
incompatible pointer type ‘struct sockaddr_in6 *’
[-Wincompatible-pointer-types]
  109 |         sa = svc_getcaller(transp);
        |            ^

Link: https://lore.kernel.org/ltp/20240617032400.298906-1-liwang@redhat.com/
Reviewed-by: Wei Gao <wegao@suse.com>
Reviewed-by: Petr Vorel <pvorel@suse.cz>
Signed-off-by: Li Wang <liwang@redhat.com>
---
 .../network/rpc/rpc-tirpc/tests_pack/rpc_svc_1/rpc_svc_1.c      | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/testcases/network/rpc/rpc-tirpc/tests_pack/rpc_svc_1/rpc_svc_1.c b/testcases/network/rpc/rpc-tirpc/tests_pack/rpc_svc_1/rpc_svc_1.c
index d1c4df97f..d367691c4 100644
--- a/testcases/network/rpc/rpc-tirpc/tests_pack/rpc_svc_1/rpc_svc_1.c
+++ b/testcases/network/rpc/rpc-tirpc/tests_pack/rpc_svc_1/rpc_svc_1.c
@@ -106,7 +106,7 @@ char *svc_getcaller_test(union u_argument *inVar, SVCXPRT * transp)
 	struct sockaddr_in *sa = NULL;
 	static int result;
 
-	sa = svc_getcaller(transp);
+	sa = (struct sockaddr_in *) svc_getcaller(transp);
 	//If the result is not NULL we consider that function call succeeds
 	//so returns 0 (PASS)
 	result = (sa != NULL) ? 0 : 1;
-- 
2.45.2

